src/helpers/ConfigHelper.ts

```diff
@@ -62,10 +62,25 @@ export class ConfigHelper {
       }
     }
   }
-
+  /**
+   * Checks if current origin matches:
+   *  - config origin exactly
+   *  - variants:
+   *    - config origin without 'www.'
+   *    - config origin with stricter protocol (we should allow http sites to initialize with https but not viceversa)
+   *    - both
+   */
   public static doesCurrentOriginMatchConfigOrigin(configOrigin: string): boolean {
     try {
-      return location.origin === new URL(configOrigin).origin;
+      const configUrl = new URL(configOrigin);
+
+      if (location.origin === configUrl.origin) {
+        return true;
+      }
+
+      const configURLWithout3W = new URL(configUrl.origin.split("www.").join(''));
+      const configOriginWithHttps = `https://${configURLWithout3W.host}`;
+      return location.origin === configOriginWithHttps;
```

Comment:
Couldn't this be simplified to?
```Typescript
return location.origin === configURLWithout3W.origin;
```

Reply:
Also `location.origin` could have `www.` in it, we should remove it from this too is compare with without `www.` for both URLs.

Reply:
For your first, I don't think that would work. We want to allow HTTPS sites (location origin) to be able to load OneSignal configured with HTTP (config origin) but not viceversa.

If we only strip 3w from the config, we could have a case where:
- `location.origin` = "https://site.com"
- `configURLWithout3W.origin` = "http://site.com"

Then your proposed simplification would break.

But if we default the comparison to always make use of the config origin with HTTPS, then we would have:
1) in the case that the location origin is HTTPS and true config is HTTP
- `location.origin` = "https://site.com"
- `configOriginWithHttps` = "https://site.com" (even though the true value isn't https)

result: match. HTTPS -> HTTP = okay

2) in the case that the location origin is HTTP and true config origin is HTTPS
- `location.origin` = "http://site.com"
- `configOriginWithHttps` = "https://site.com" (matches true config value)

result: doesn't match. HTTP -> HTTPS = insecure

So the idea here is to strip 3w and then make a unidirectional comparison that only allows HTTPS->HTTP.

Per your second comment, I agree we should remove from both URLs